homeassistant/components/vesync/number.py

```diff
@@ -0,0 +1,114 @@
+"""Support for number settings on VeSync devices."""
+import logging
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .common import VeSyncDevice
+from .const import DOMAIN, VS_DISCOVERY, VS_DISPATCHERS, VS_NUMBERS
+
+_LOGGER = logging.getLogger(__name__)
+
+DEV_TYPE_TO_HA = {
+    "Classic300S": "humidifier",
+}
+
+MAX_MIST_LEVEL = 9
+MIN_MIST_LEVEL = 1
+MIST_LEVEL_STEP = 1
+
+MAX_NIGHT_LIGHT_BRIGHTNESS = 100
```

Comment:
Can we expose the night light as a Light entity tied to the humidifier device rather than as a Number? Allows use of existing brightness support.

Reply:
Sure, that makes sense.